docker-compose.yml

```diff
@@ -0,0 +1,22 @@
+version: "3.3"
+services:
+  stationeers:
+    container_name: stationeers # Change me if needed
+#    stdin_open: true # docker run -i - uncomment for testing
+#    tty: true # docker run -t - uncomment for testing
+    ports:
+      - "27500:27500/udp" # Game Port - default 27500/udp
+      - "27500:27500/tcp" # Web RCON Port - Same as Game port - default 27500/tcp
+      - "27015:27015/udp" # Steam Update Port - default 27015/udp
+    image: hetsh/stationeers:latest # For Modded servers: Best you define the exact label.
+    environment:
+      - WORLD_TYPE=Mars
+      - WORLD_NAME=mars_save
+#      - SAVE_INTERVAL=300 # Uncomment to change the save interval - default =300
+#      - CLEAR_INTERVAL=-1 # Uncomment to set the idle player clearing to be disabled.
+#      - SERVER_OPTS=-bindip x.x.x.x # Uncomment to set an IP assigned via ZeroTier running on the same docker host.
```

Comment:
> Binding to a ZeroTier IP is very specific and not straight forward. This might be a superb wiki entry though, what do you think?

Its actually _very_ simple.  I'm not thinking about 6PLANE though - **that** is out of the scope of even this repo.

Hard (manual) way - https://zerotier.atlassian.net/wiki/spaces/SD/pages/7536656/Running+ZeroTier+in+a+Docker+Container

Easier ( _example_ https://hub.docker.com/r/zyclonite/zerotier/ ) way;
```
docker run --name zerotier-one --device=/dev/net/tun --net=host \
  --cap-add=NET_ADMIN --cap-add=SYS_ADMIN \
  -v /var/lib/zerotier-one:/var/lib/zerotier-one zyclonite/zerotier
```

There are quite a few already made docker containers for zero-tier.  That was just _one_ example.



> Specifying SERVER_OPTS would have the same effect as just adding them as [commands](https://docs.docker.com/compose/compose-file/compose-file-v3/#command) right? So it should be possible to remove the SERVER_OPTS completely.

Not with the way that you have done your docker.  Yes you can change the entry point command after the fact but not on initial up/start.

And because you have an env var of `$SERVER_OPTS` its easy to define/add without the need to modify/re-spin the container, if you have a template in place with it, etc.

Reply:
> Its actually very simple. I'm not thinking about 6PLANE though - that is out of the scope of even this repo.

Still, binding to a specific IP is not part of a standard setup. We should not include it in the example. I would rather prefer an entry in the Wiki.

> Not with the way that you have done your docker. Yes you can change the entry point command after the fact but not on initial up/start.

I think you are mixing up [ENTRYPOINT](https://docs.docker.com/engine/reference/builder/#entrypoint) with [CMD](https://docs.docker.com/engine/reference/builder/#cmd). You can use CMD to add parameters to an ENTRYPOINT. Exactly what SERVER_OPTS is currently used for. I opened issue #6 which we can resolve after merging the changes.

Reply:
>Still, binding to a specific IP is not part of a standard setup. We should not include it in the example. I would rather prefer an entry in the Wiki.

Actully it **is**. 

```
rocketstation_DedicatedServer.x86_64" -batchmode -nographics -autostart -basedirectory="$DATA_DIR" -logfile="$LOG_DIR/game.log" -autosaveinterval="$SAVE_INTERVAL" -clearallinterval="$CLEAR_INTERVAL" -worldtype="$WORLD_TYPE" -worldname="$WORLD_NAME" -loadworld="$WORLD_NAME" $SERVER_OPTS
```

To actually **bind** stationeers to an ip address when its running on a system with access to multi VLANS', etc you **must** use `-bind`  as that _binds the server application to the ip_.

You are confusing between docker itself and with how unity & stationeers can bind to an ip address.